agent/balrogagent/cmd.py

```diff
@@ -43,7 +43,21 @@
                     else:
                         logging.debug("Unknown change type!")
 
-                    # If it *is* ready, enact it!
+                    # If it *is* ready, check if all the required signoffs have
+                    # been obtained
+                    required_signoffs = change.get('required_signoffs') or {}
+                    signoff_list = change.get('signoffs') or {}
+                    obtained_signoffs = {}
+                    for user, role in signoff_list.items():
+                        if obtained_signoffs.get(role) is not None:
+                            obtained_signoffs[role] += 1
+                        else:
+                            obtained_signoffs[role] = 1
+                    for role, number in required_signoffs.items():
+                        if required_signoffs.get(role, 0) > obtained_signoffs.get(role, 0):
+                            logging.debug("required signoffs not obtained, aborting")
+                            ready = False
+                    # If we have all required signoffs, go ahead
```

Comment:
This logic looks fine, but it belongs in a separate function. I also wonder if it would be better to just re-use https://github.com/mozilla/balrog/blob/ac29e19c2d3e4f29af74296d03c908f0ff32320b/auslib/db.py#L132 ?